boa/src/builtins/value/mod.rs

```diff
@@ -149,6 +150,95 @@ impl Value {
+                    new_obj.properties.insert(
+                        idx.to_string(),
+                        Property::default()
+                            .value(Self::from_json(json.clone(), interpreter))
+                            .writable(true)
+                            .configurable(true),
+                    );
+                }
+                new_obj.properties.insert(
+                    "length".to_string(),
+                    Property::default().value(Self::from(vs.len())),
+                );
+                Self::object(new_obj)
+            }
+            JSONValue::Object(obj) => {
+                let new_obj = Value::new_object(Some(&interpreter.realm.global_obj));
+                for (key, json) in obj.iter() {
+                    let value = Self::from_json(json.clone(), interpreter);
+                    new_obj.set_property(
+                        key.clone(),
+                        Property::default()
+                            .value(value)
+                            .writable(true)
+                            .configurable(true),
+                    );
+                }
+                new_obj
+            }
+            JSONValue::Null => Self::null(),
+        }
+    }
+
+    /// Conversts the `Value` to `JSON`.
+    pub fn to_json(&self) -> JSONValue {
+        match *self.data() {
+            ValueData::Null => JSONValue::Null,
+            ValueData::Boolean(b) => JSONValue::Bool(b),
+            ValueData::Object(ref obj) => {
+                if obj.borrow().kind == ObjectKind::Array {
+                    let mut arr: Vec<JSONValue> = Vec::new();
+                    obj.borrow().properties.keys().for_each(|k| {
+                        if k != "length" {
+                            let value = self.get_field(k.to_string());
+                            if value.is_undefined() || value.is_function() {
+                                arr.push(JSONValue::Null);
+                            } else {
+                                arr.push(self.get_field(k.to_string()).to_json());
+                            }
+                        }
+                    });
+                    JSONValue::Array(arr)
+                } else {
+                    let mut new_obj = Map::new();
+                    obj.borrow().properties.keys().for_each(|k| {
+                        let key = k.clone();
+                        let value = self.get_field(k.to_string());
+                        if !value.is_undefined() && !value.is_function() {
+                            new_obj.insert(key, value.to_json());
+                        }
+                    });
+                    JSONValue::Object(new_obj)
+                }
+            }
+            ValueData::String(ref str) => JSONValue::String(str.clone()),
+            ValueData::Rational(num) => JSONValue::Number(
+                JSONNumber::from_f64(num).expect("Could not convert to JSONNumber"),
```

Comment:
I'm not 100% comfortable with this panic. What if the number is infinite or NaN?

Reply:
I'm not sure. I'll have to check the spec. Most of this PR was just pulled over from previous implementation.

Reply:
I'm having a hard time finding anything about it in the spec. Popping open the dev tools in Firefox gives me results, but I can't find any mention of the behavior that I'm seeing in the spec. Here is what Firefox provides:
```javascript
JSON.stringify({ x: NaN }) // "{\"x\":null}"
JSON.stringify({ x: Number.EPSILON }) // "{\"x\":2.220446049250313e-16}"
JSON.stringify({ x: Number.MAX_SAFE_INTEGER }) // "{\"x\":9007199254740991}"
JSON.stringify({ x: Number.MIN_SAFE_INTEGER }) // "{\"x\":-9007199254740991}"
JSON.stringify({ x: Number.MAX_VALUE }) // "{\"x\":1.7976931348623157e+308}"
JSON.stringify({ x: Number.MIN_VALUE }) // "{\"x\":5e-324}"
JSON.stringify({ x: Number.POSITIVE_INFINITY }) // "{\"x\":null}"
JSON.stringify({ x: Number.NEGATIVE_INFINITY }) // "{\"x\":null}"
```

Reply:
I would venture to guess that it is buried in [SerializeJSONProperty](https://tc39.es/ecma262/#sec-serializejsonproperty).

The JSON object could probably benefit from a stricter adherence to the implementation described in the spec. It seems that JSON object might follow the intent of the spec, but not the implementation.

Reply:
OK, so I would suggest using `null` as default instead of panicking, and I'll try to find why is Firefox returning that.

Reply:
returning null. :heavy_check_mark: